### PR TITLE
fix: reduce number of calls made to infura/alchemy

### PR DESCRIPTION
**Pull request by ezhou7**
# Task:
https://github.com/opynfinance/squeeth-monorepo/issues/11

## Description

Looks like calling `useController()` in many different places is causing repeat API calls to Infura/Alchemy. This PR attempts to reduce the number of API calls by reducing the number of `useController()` calls.

Ideally this alone would fix issue 11, but it alone cannot. I think there are two places where the API calls really skyrocket from `useController()` - inside `packages/frontend/src/hooks/usePositions.ts` there are two function `usePositions` and `useShortPositions`. Replacing the `const { getDebtAmount } = useController()` with a dummy method reduces a lot of API calls, which leads me to suspect that if the `getDebtAmount` method could be extracted out from `useController()`, a lot of API calls could be removed.

Fixes # (issue)
https://github.com/opynfinance/squeeth-monorepo/issues/11

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Document update

## How Has This Been Tested

Start the frontend for `main` branch and record the number of Infura/Alchemy API calls.

Start the frontend for `issue-11` branch and open the networks tab to compare the new number of Infura/Alchemy API calls. There should be fewer API calls.

Please reproduce on your end as this is my first PR for the Squeeth FE and I am not so familiar with the codebase. Thanks!

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] Added video recordings if it is a UI change

### CODE REVIEW

**Comment by EstherSeyi** [comment]
https://user-images.githubusercontent.com/40301524/152754900-41c0cb98-1176-483f-9f61-c330f9756b9a.mp4

The preview seems to be erroring when I open it on my browser.

**Comment by yichiehliu** [comment]
> 

Same issue for me

**Comment by EstherSeyi** [comment]
I'm still having the same issue with opening the preview on my browser.

**Comment by akwritescode** [comment]
hey @EstherSeyi 
I dont think the previw got updated after he pushed.
Just authorized vercel.
Let's see after this build is done https://vercel.com/opynfinance/continuouscall/FBMRKBu6N3MiZmLjYFH3nqqf3Vuh

**Comment by akwritescode** [comment]
ok it got deployed now. i think he still needs to fix some issues.
Let's wait till the issues are addressed before reviewing the PR

**Comment by ezhou7** [comment]
Network calls to Alchemy for this PR Preview:
PR networks tab


Network calls to Alchemy for Prod UI:
Prod networks tab

**Comment by KMKoushik** [comment]
LGTM

@EstherSeyi @yichiehliu @alexisgauba You have any thoughts

File: packages/frontend/package.json [hunk 1]
```diff
@@ -32,9 +32,11 @@
     "framer-motion": "^5.3.3",
     "graphql": "^16.1.0",
     "gray-matter": "^4.0.3",
+    "jotai": "^1.5.3",
     "kaktana-react-lightweight-charts": "^2.0.0",
     "markdown-to-jsx": "^7.1.5",
     "next": "^12.0.7",
+    "particule": "^0.0.4",
```
Comment (EstherSeyi) [inline packages/frontend/package.json:39]: we could uninstall this since we're no longer using it.

**Comment by EstherSeyi** [comment]
The result does not seem to be consistent, sometimes the number of calls is actually lower and other times it's like below.
This preview
Screenshot 2022-02-14 at 09 58 59

Prod
Screenshot 2022-02-14 at 09 59 09

File: packages/frontend/package.json (hunk 1 above)
Comment (ezhou7) [reply packages/frontend/package.json:39]: Ty for the catch

**Comment by ezhou7** [comment]
@EstherSeyi Can you check the number of `POST` calls to Alchemy API b/w this PR preview and prod? Would like to know what the difference in calls is for you in that aspect.

**Comment by EstherSeyi** [comment]
> @EstherSeyi Can you check the number of `POST` calls to Alchemy API b/w this PR preview and prod? Would like to know what the difference in calls is for you in that aspect.

Okay, I checked for the POST requests to Alchemy API and it looks good. I got 11 calls on average on the PR as against 40 calls in prod.

**Review by akwritescode — APPROVED** [review]
lgtm

Please fix the merge conflicts @ezhou7

**Review by yichiehliu — APPROVED** [review]

**Review by EstherSeyi — APPROVED** [review]

**Comment by KMKoushik** [comment]
@ezhou7 Great work! Thanks!

**Pull request merged**